test/conformance/helpers/broker_data_plane_test_helper.go

```diff
@@ -0,0 +1,363 @@
+			body := fmt.Sprintf(`{"msg":%q}`, eventID)
+			responseSink := "http://" + client.GetServiceHost(loggerName)
+			client.SendRequestToAddressable("twohundred-test-sender", broker.Name, testlib.BrokerTypeMeta,
+				map[string]string{
+					"ce-specversion": "1.0",
+					"ce-type":        testlib.DefaultEventType,
+					"ce-source":      "2XX on good CE",
+					"ce-id":          eventID,
+					"content-type":   ce.ApplicationJSON,
+				},
+				body,
+				sender.WithResponseSink(responseSink),
+			)
+
+			eventTracker.AssertExact(1, recordevents.MatchEvent(sender.MatchStatusCode(202))) // should probably be a range
+
+		})
+		//Respond with 400 on bad CE
+		t.Run("Repsond with 400 on bad CE", func(t *testing.T) {
+			eventID := "four-hundred-on-bad-ce"
+			body := ";la}{kjsdf;oai2095{}{}8234092349807asdfashdf"
+			responseSink := "http://" + client.GetServiceHost(loggerName)
+			client.SendRequestToAddressable("fourhundres-test-sender", broker.Name, testlib.BrokerTypeMeta,
+				map[string]string{
+					"ce-specversion": "9000.1", //its over 9,000!
+					"ce-type":        testlib.DefaultEventType,
+					"ce-source":      "400 on bad CE",
+					"ce-id":          eventID,
+					"content-type":   ce.ApplicationJSON,
+				},
+				body,
+				sender.WithResponseSink(responseSink))
+			eventTracker.AssertExact(1, recordevents.MatchEvent(sender.MatchStatusCode(400)))
+		})
+
+	})
+}
+
+//At consumer
+//No upgrade of version
+//Attributes received should be the same as produced (attributes may be added)
+//Events are filtered
+//Events are delivered to multiple subscribers
+//Deliveries succeed at least once
+//Replies are accepted and delivered
+//Replies that are unsuccessfully forwarded cause initial message to be redelivered (Very difficult to test, can be ignored)
+func BrokerV1Beta1ConsumerDataPlaneTestHelperWithChannelTestRunner(
+	t *testing.T,
+	brokerClass string,
+	channelTestRunner testlib.ComponentsTestRunner,
+	setupClient ...testlib.SetupClientOption,
+) {
+	channelTestRunner.RunTests(t, testlib.FeatureBasic, func(t *testing.T, channel metav1.TypeMeta) {
+		client := lib.Setup(t, true, setupClient...)
+		defer lib.TearDown(client)
+
+		brokerName := "br"
+		triggerName := "trigger"
+		secondTriggerName := "second-trigger"
+		loggerName := "logger-pod"
+		secondLoggerName := "second-logger-pod"
+		eventTracker, _ := recordevents.StartEventRecordOrFail(client, loggerName)
+		defer eventTracker.Cleanup()
+		secondTracker, _ := recordevents.StartEventRecordOrFail(client, secondLoggerName)
+		defer secondTracker.Cleanup()
+		client.WaitForAllTestResourcesReadyOrFail()
+
+		config := client.CreateBrokerConfigMapOrFail(brokerName, &channel)
+
+		broker := client.CreateBrokerV1Beta1OrFail(
+			brokerName,
+			resources.WithBrokerClassForBrokerV1Beta1(brokerClass),
+			resources.WithConfigForBrokerV1Beta1(config),
+		)
+		client.WaitForResourceReadyOrFail(broker.Name, testlib.BrokerTypeMeta)
+
+		trigger := client.CreateTriggerOrFailV1Beta1(
```

Comment:
Then, I think if you do what I propose above, these (at least seemingly) identical tests can just be called by calling that method?